testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/CommonDeprecations.groovy

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.fixtures
+
+import groovy.transform.SelfType
+import org.gradle.integtests.fixtures.executer.GradleContextualExecuter
+
+/**
+ * Apply this trait to tests that may need to expect common deprecations.
+ */
+@SelfType(HasGradleExecutor)
+trait CommonDeprecations {
```

Comment:
:x: `CommonDeprecations` as an aspiration but right now this is only about `Task#getProject` so why not name it more explicitly? We could also make it about `StableConfigurationCacheDeprecations`.

Reply:
Nice, that is general/specific enough for me.